### PR TITLE
fix streams byob request handling after close for spec compliance

### PR DESCRIPTION
**Pull request by anonrig**
Fix ReadableByteStreamController to properly handle byob requests after close(), making respondWithNewView() throw correct errors per the WHATWG Streams spec.

Removed premature byob request invalidation because spec required byob requests to remain accessible after close so that respondWithNewView throw correct errors. Had to store original buffer byte length and byte offset as well in order to make it available for validation even after the underlying ByobRequest is invalidated during close.

Had to update several test contexts due to invalid configuration.

Fixes WPT readable-byte-streams/bad-buffers-and-views.any.js

### CODE REVIEW

**Comment by jasnell** [comment]
I might merge this into the state-machine update for this class. Please do not merge this yet.

**Comment by anonrig** [comment]
> I might merge this into the state-machine update for this class. Please do not merge this yet.

@jasnell ping

**Comment by jasnell** [comment]
No updates yet, the remaining state machine improvements are pending completion of rolling out a couple autogates. One of those just finished rolling out this morning the remaining one will roll out monday morning. I'll update when there's additional progress.

**Comment by jasnell** [comment]
Ok, the state machine changes have landed. This should be revisited now.

**Comment by anonrig** [comment]
@jasnell you can re-review this. i've fixed the conflicts.

File: src/workerd/api/streams/standard.c++ [hunk 1]
```diff
@@ -2348,7 +2351,35 @@ void ReadableStreamBYOBRequest::respondWithNewView(jsg::Lock& js, jsg::BufferSou
     if (!controller.canCloseOrEnqueue()) {
       JSG_REQUIRE(view.size() == 0, TypeError,
           "The view byte length must be zero after the stream is closed.");
-      KJ_ASSERT(impl.readRequest->isInvalidated());
+
+      if (FeatureFlags::get(js).getPedanticWpt()) {
```
Comment (jasnell) [inline src/workerd/api/streams/standard.c++:2355]: nit (optional): The `view.size() == 0` check on line 2352 runs _before_ this `pedanticWpt` block, so a detached buffer with a non-zero view size would hit the size `TypeError` before the detached-buffer check. The WHATWG Streams spec may expect the detached check to come first in the closed-state path — WPT tests passing is strong evidence the current ordering works for tested cases, but it might be worth a quick cross-check against the spec step ordering to confirm there are no uncovered edge cases where the error type would differ.

Likely fine as-is — just flagging for awareness.

File: src/workerd/api/streams/queue.c++ [hunk 2]
```diff
@@ -872,15 +877,38 @@ v8::Local<v8::Uint8Array> ByteQueue::ByobRequest::getView(jsg::Lock& js) {
   return v8::Local<v8::Uint8Array>();
 }
 
+size_t ByteQueue::ByobRequest::getOriginalBufferByteLength(jsg::Lock& js) const {
+  KJ_IF_SOME(req, request) {
+    KJ_IF_SOME(size, req.pullInto.store.underlyingArrayBufferSize(js)) {
+      return size;
+    }
+  }
+  return 0;
```
Comment (jasnell) [inline src/workerd/api/streams/queue.c++:886]: nit (optional): `return 0` here is ambiguous — it's the same value as a legitimately zero-length buffer. In practice this shouldn't matter because the snapshot is taken in the `Impl` constructor when the request is always valid, but the API is a bit fragile.

Two possible improvements (pick one, or neither):
1. Use `kj::Maybe<size_t>` as the return type to distinguish "not available" from "zero".
2. Add a `KJ_ASSERT` in the `Impl` constructor to assert the snapshotted values are valid at snapshot time, making the invariant explicit.

Probably not worth changing if you're happy with the current shape — just noting it.

File: src/workerd/api/streams/standard.c++ (hunk 1 above)
Comment (anonrig) [reply src/workerd/api/streams/standard.c++:2355]: I think we should add a rule to AI assistant rules to avoid using `--` lol

File: src/workerd/api/streams/queue.c++ (hunk 2 above)
Comment (anonrig) [reply src/workerd/api/streams/queue.c++:886]: This looks good to me. I'll add this.

Comment (anonrig) [reply src/workerd/api/streams/queue.c++:886]: Actually this might be a breaking change. I'll look into it once we land this.

File: src/workerd/api/streams/standard.c++ (hunk 1 above)
Comment (jasnell) [reply src/workerd/api/streams/standard.c++:2355]: emdashes are awesome. it's probably one of the only great things AI agents truly get right.
